### PR TITLE
Fix various unit test issues so that tests pass reliably

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: src/github.com/getlantern/balancer/balancer_test.go [hunk 1]
```diff
@@ -86,48 +86,6 @@ func TestRandomDialer(t *testing.T) {
 	assertWithinRangeOf(t, atomic.LoadInt32(&d3Attempts), 1000, 100)
 }
 
-func TestLoadBalancing(t *testing.T) {
```
Comment (oxtoacart) [inline src/github.com/getlantern/balancer/balancer_test.go]: @fffw Indicated that this test is not necessary, so I removed it.

File: src/github.com/getlantern/idletiming/idletiming_test.go [hunk 2]
```diff
@@ -18,8 +18,8 @@ var (
 
 	clientTimeout                 = 25 * time.Millisecond
 	serverTimeout                 = 10 * clientTimeout
-	slightlyLessThanClientTimeout = time.Duration(int64(float64(clientTimeout.Nanoseconds()) * 0.9))
-	slightlyMoreThanClientTimeout = time.Duration(int64(float64(clientTimeout.Nanoseconds()) * 1.1))
+	slightlyLessThanClientTimeout = time.Duration(int64(float64(clientTimeout.Nanoseconds()) * 0.5))
+	slightlyMoreThanClientTimeout = time.Duration(int64(float64(clientTimeout.Nanoseconds()) * 1.5))
```
Comment (oxtoacart) [inline src/github.com/getlantern/idletiming/idletiming_test.go]: Sometimes the margin we provided wasn't enough, so I bumped it up.

File: src/golang.org/x/net/.gitattributes [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Treat all files in this repo as binary, with no git magic updating
```
Comment (oxtoacart) [inline src/golang.org/x/net/.gitattributes]: We need the net/proxy package for the lantern test.

**Comment by myleshorton** [comment]
LGTM @oxtoacart  thanks! Weird all those golang files didn't get pulled in before -- I thought they did. Did just certain packages not make it over with the dependency update PR?

**Comment by oxtoacart** [comment]
I think I may have missed that test dependency because a prior test failed and I never got this test failure.  But I'm not sure.

**Pull request merged**